src/librbd/api/Trash.cc

```diff
@@ -116,27 +118,18 @@ int enable_mirroring(IoCtx &io_ctx, const std::string &image_id) {
 } // anonymous namespace
 
 template <typename I>
-int Trash<I>::move(librados::IoCtx &io_ctx, rbd_trash_image_source_t source,
-                   const std::string &image_name, uint64_t delay) {
+int Trash<I>::move_by_id(librados::IoCtx &io_ctx, rbd_trash_image_source_t source,
+                   const std::string &image_id, const std::string &image_name,
+                   uint64_t delay, bool check_for_watchers) {
```

Comment:
nit: as the function also includes `image_name` parameter and may work when `image_id` is empty (and `image_name` is not), then probably name this function `move`? It will differ from another `move` by signature.

Reply:
actually I fine with `move_by_id` too, so just ignore if you don't like it.

Reply:
I think it goes with the existing convention : `open` and `open_by_id` . Will keep it